packages/logger/src/__tests__/index.test.ts

```diff
@@ -0,0 +1,138 @@
+import { PassThrough } from "stream";
+import * as winston from "winston";
+import WinstonTransport from "winston-transport";
+import * as bunyan from "bunyan";
+import * as loglevel from "loglevel";
+import * as log4js from "log4js";
+import type { Logger } from "..";
+
+const LOWEST_LOG_LEVEL = "debug";
+const KNOWN_DEBUG_MESSAGE = "This is a debug message";
+
+// For typechecking the logger in question's compatibility
+// with the Logger interface.
+interface HasLogger {
+  logger: Logger;
+}
+
+describe("Logger interface compatibility", () => {
+  it("with loglevel", () => {
+    const sink = jest.fn();
+    const logger = loglevel.getLogger("test-logger-loglevel");
+
+    logger.methodFactory =
+      (_methodName, level): loglevel.LoggingMethod =>
+      (message) =>
+        sink({ level, message });
+
+    // The `setLevel` method must be called after overwriting `methodFactory`.
+    // This is an intentional API design pattern of the loglevel package:
+    // https://www.npmjs.com/package/loglevel#writing-plugins
+    logger.setLevel(loglevel.levels.DEBUG);
+
+    // type checks Logger interface compatibility
+    const hasLogger: HasLogger = { logger };
```

Comment:
do you get why we do this rather than `const loggerInterface: Logger = logger`?

Reply:
Yeah, simplified this a bit in 911dda3.

There's still an extra step since I can't directly create a logger instance as a `Logger` like so:
`const logger: Logger = log4js.getLogger();`

...but the interface isn't necessary after all.